sklearn_hierarchical_classification/classifier.py

```diff
@@ -163,7 +178,14 @@ def fit(self, X, y=None, sample_weight=None):
         self
 
         """
-        X, y = check_X_y(X, y, accept_sparse='csr')
+        if self.preprocessing:
```

Comment:
@fbenites can you briefly explain why we need to divide into cases based on `preprocessing` here? e.g. what is the difference for the sake of X, y input validation?

Reply:
The point is that X here may not been vectorized (like in unprocessed text form). with preprocessing enable, it will be vectorized later (the idea is even to use a gridsearch to optimize the preprocessing per node tree).